### DISCUSSION TITLE
Q: Possible ways to limit search results number

### DISCUSSION THREAD

**Discussion by zhzLuke96** [searxng/searxng#2240] [Q&A] [answered]
I have a question regarding the search API interface of Searxng.

Specifically, I was wondering if it is possible to limit the number of search results returned by the API.
For instance, is it possible to only retrieve the first result from each search engine or just the top result overall?

I have already gone through the API documentation, but it seems that there is no such parameter available. Do you have any suggestions on how I can achieve this? Should I take a look at the source code?

**Comment by return42** [comment, marked as answer]
This is not provided in SearXNG.  The engines paginate in pages, the number of hits per page depends on the origin search-engine for which an SearXNG-engine is implemented.

If it is about performance: Usually the round trip for a query to a search engine is not significantly shorter, if you only want to get the first hit (but this is not possible anyway, because the search engines do not offer this function, as mentioned above).
